clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumerImpl.java

```diff
@@ -456,7 +443,8 @@ private enum AcknowledgementMode {
                       final ConsumerMetadata metadata,
                       final int requestTimeoutMs,
                       final int defaultApiTimeoutMs,
-                      final String groupId) {
+                      final String groupId,
+                      final String acknowledgementMode) {
```

Comment:
I'd probably give this a different variable name, such as `acknowledgementModeConfig`.